### PR TITLE
Handle Deprecation Warnings

### PR DESCRIPTION
**Pull request by shtakai**
https://github.com/akabiru/fakerbot/issues/7

- Display as `deprecate` if respond to `deprecate_...`

### CODE REVIEW

**Comment by shtakai** [comment]
@akabiru `CODEBEAT` said some complaints, so that I refactored a little. 
But not fixed completely...
```
Assignment Branch Condition too high   FakerBot::Renderer#verbose_output
infoabc = 11.18
```

File: lib/fakerbot/renderer.rb [hunk 1]
```diff
@@ -73,5 +64,20 @@ def gt_screen_height?(data_tree)
...
+      arr << crayon.dim.white("=> #{fake.to_s}") << crayon.dim.magenta.bold("#{message}")
+    end
+
+    def faker_method(method, const)
+      [const.public_send(method), deprecate(method, const)]
+    rescue ArgumentError
+      ['N/A', '']
+    end
+
+    def deprecate(method, const)
```
Comment (akabiru) [inline lib/fakerbot/renderer.rb]: What are your thoughts on renaming this method to `ensure_method_is_supported` or sth similar for clarity?

File: lib/fakerbot/renderer.rb [hunk 2]
```diff
@@ -54,16 +54,7 @@ def leaf(const, methods)
...
-        if verbose?
-          fake = begin
-            const.public_send(method)
-          rescue ArgumentError
-            'N/A'
-          end
-          arr << crayon.dim.white("=> #{fake.to_s}")
-        end
-      end
+      [method.to_s].tap { |arr| verbose_output(method, const, arr) if verbose? }
```
Comment (akabiru) [inline lib/fakerbot/renderer.rb:57]: ⭐️ I like the use of extract method here! 👍

File: lib/fakerbot/renderer.rb [hunk 3]
```diff
@@ -73,5 +64,20 @@ def gt_screen_height?(data_tree)
...
+    end
+
+    def faker_method(method, const)
+      [const.public_send(method), deprecate(method, const)]
+    rescue ArgumentError
+      ['N/A', '']
+    end
+
+    def deprecate(method, const)
+      const.respond_to?("_deprecated_#{method.to_s}".to_sym) ? ' ( WILL BE DEPRECATED )' : ''
```
Comment (akabiru) [inline lib/fakerbot/renderer.rb]: Can we use direct symbol interpolation here? `:"_deprecated_#{method}"`?

**Review by akabiru — COMMENTED** [review]
This looks great @shtakai ! Just some few questions/suggestions. 🙂

File: lib/fakerbot/renderer.rb (hunk 1 above)
Comment (shtakai) [reply lib/fakerbot/renderer.rb]: I couldn't pick up `ensure_..`  from my brain. 😮 
I remembered!

File: lib/fakerbot/renderer.rb (hunk 3 above)
Comment (shtakai) [reply lib/fakerbot/renderer.rb]: Thank you. 😄

**Review by akabiru — APPROVED** [review]
This looks great to me. Thanks! 🎉

**Pull request merged**